torch/csrc/jit/codegen/cuda/partition.cpp

```diff
@@ -84,33 +71,37 @@ bool isFusableCudaFusionGroup(const Node* const node) {
 bool isFusableCudaFusionGroup(
     const Node* const fusion,
     const Node* const node) {
-  if (isFusableNode(node)) {
-    auto device = getDevice(fusion);
+  if (isFusableCudaFusionGroup(node)) {
+    // TODO: ensure legit fusion.
+    // issue 0: currently codegen doesn't support broadcasting, except in the
+    //          form of stride 0.
+    // We WAR by explicitly extend all tensor to be the broadcasted size. This
+    // however requires that we have identical tensor shape for all output
```

Comment:
Is this now checked in the shape propagation?

Reply:
No this is currently not checked anywhere.
Checking shape is tricky for two reasons: 1. we could ended up being too conservative on not taking nodes that we should be able to fuse; 2. currently if we are not using profiling executor, there's no guarantee that shape information will be present.

Right now we just blindly accepts nodes in fusion group. If we ended up not being able to generate kernel, we'll fallback to normal interpreter and run it without codegen. (This is the same bahavior legacy fuser has and would make our functional parity comparison easier).